Added missing endpoints and minor clean-up.

abelsm2 opened the pull request:
Added the following endpoints:
- /time_attack/member_season_results
- /stats/member_recap
- /stats/member_division
- /series/past_seasons
- /member/participation_credits
- /member/awards
- /league/roster

Removed the check to ensure a cust_id was passed to the following methods because they all are supposed to default to the authenticated user and the cust_id is not required.
- `stats_member_summary()`
- `stats_member_yearly()`
- `stats_member_recent_races()`

There were other minor updates, including revising the incorrectly documented return type in `league_season_sessions()` to eliminate issues with inferred return types when using the library.

jasondilworth56 commented:
This looks good on a quick look through, give me a little bit just to catch any issues and then we'll get it done.

jasondilworth56 commented on src/iracingdataapi/client.py:
```diff
@@ -821,11 +861,19 @@ def member_info(self):
         """
         return self._get_resource("/data/member/info")
 
+    def member_participation_credits(self):
```
I've avoided adding any that are specific only to the authenticated member, and I'm still erring on the side of this probably being unnecessary for this wrapper, but I'm up for being convinced.

jasondilworth56 commented on src/iracingdataapi/client.py:
```diff
@@ -1204,10 +1305,25 @@ def get_series_assets(self):
+            dict: a dict containing information about the series and a list of seasons.
+        """
+        payload = {"series_id": series_id}
+        return self._get_resource("/data/series/past_seasons", payload=payload)
```
I think it could be good to only return the `series` part of the response, having checked for `success` beforehand. Similar to how you did in `member_awards`
```suggestion
        resource = self._get_resource("/data/series/past_seasons", payload=payload)
        
        if resource.get('data', {}).get('success'):
            return resource['series']
        else:
            return {}
```

Also probably better to return something more insightful when there's an unsuccessful call.

abelsm2 replied:
Funny, I was just running into this while trying to implement this in my bot.  Good idea.  I'll push an update later today when I can get to it.

As far as the return value in an unsuccessful call... I'm not really sure how the /data API works.  With `member_awards`, if I try a random number as garbage input I get an empty list in return and a value of 'success', so my check for `success` is probably not really needed.  With `series_past_seasons`, if I try garbage input I get a RuntimeError for an Unhandled Non-200 response (Response [404]) and I'm not really sure how to handle that.

I could probably just pull the check from both methods and leave error handling to the user to ensure the input is valid.

abelsm2 commented:
- Caught 1 more instance of throwing an exception when it should default to the authenticated user in `stats_member_career`.
- Adjusted the return value on `series_past_seasons` as suggested.
- Removed the error checks I had since it's not documented when the /data API would actually return a value other than `success` and I think it's better left to the user to handle any errors from invalid inputs.
- Removed all methods that I created that only work for the authenticated user.  Note: `member_info` already existed and only works for the authenticated user, but I can see use cases for that so have not suggested deprecating that method.

jasondilworth56 commented:
All merged and packaged up to PyPi, thanks @abelsm2!
